modules/core/server/services/influx.server.service.js

```diff
@@ -44,12 +46,12 @@ exports.getClient = function(callback) {
  * influxdb (camelCase)
  * @param {Object} fields - key: value pairs will be saved in influxdb as field
  * key: field value
- * @param {number|Date} [fields.time=new Date()] - time of measurement with precision ms
+ * @param {number|Date} [fields.time=new Date()] - time of measurement with precision nanosecond
```

Comment:
Based on the description of [writePoints](https://node-influx.github.io/class/src/index.js~InfluxDB.html#instance-method-writePoints) method, passing javascript `Date` is fine. However, the messageToInflux service is indeed using timestamp (number) with millisecond precision (see the [related test](/Trustroots/trustroots/blob/upgrade-node-influx-v5/modules/messages/tests/server/message-to-influx.server.service.tests.js#L240)). It's not clear to me how the defaults of node-influx work now.

Note1: `influx.writeMeasurement` uses `writePoints` method internally.
Note2: there is a lot of space for improving that linked test, i can see.
Note3: To remove confusion, we could require `Date` type for time or [explicitly provide the precision](/Trustroots/trustroots/blob/upgrade-node-influx-v5/modules/core/server/services/influx.server.service.js#L79) as the 3rd parameter of [`writeMeasurement`](https://node-influx.github.io/class/src/index.js~InfluxDB.html#instance-method-writeMeasurement). Node-influx docs [suggest the precision to be as rough as possible](https://node-influx.github.io/typedef/index.html#static-typedef-IWriteOptions) for efficiency. I doubt we need nanosecond precision.